### PR TITLE
Module - User Management

### PR DESCRIPTION
**Pull request by rakeshar3796**

### CODE REVIEW

File: web/src/page/admin/admin.module.css [hunk 1]
```diff
@@ -1,16 +1,48 @@
...
-    justify-content: center;
+  height: 100%;
+  display: flex;
+  align-items: initial;
+  justify-content: center;
 }
 
 .userTable {
-    width: 98%;
+  width: 98%;
```
Comment (rakeshar3796) [inline web/src/page/admin/admin.module.css:9]: width: 100%;
padding: 0 16px;

File: web/src/page/admin/index.tsx [hunk 2]
```diff
@@ -1,104 +1,309 @@
...
+  function onCreateuserClick() {
+    form.resetFields();
+    setModalState(MODAL_STATES.CREATE);
+  }
+
+  const onFormSubmit = (values: IUser) => {
+    function updateData(response: IUser) {
+      setUserData((state: IUserList) => {
+        if (modalState === MODAL_STATES.CREATE) {
+          return [...state, response];
```
Comment (rakeshar3796) [inline web/src/page/admin/index.tsx:53]: check this

File: web/src/page/admin/index.tsx [hunk 3]
```diff
@@ -1,104 +1,309 @@
...
-          onConfirm={confirm}
-          // onCancel={cancel}
-          okText="Delete"
-          cancelText="Cancel"
-        >
-          <a href="/" className={styles.userDelete}>Delete</a>
-        </Popconfirm>
+          const cloneData = { ...values };
+          state[dataIndex] = cloneData;
+          state[dataIndex].roles = [cloneData.roles].flat();
```
Comment (rakeshar3796) [inline web/src/page/admin/index.tsx:60]: check this

File: web/src/page/admin/index.tsx [hunk 4]
```diff
@@ -1,104 +1,309 @@
...
+      url = `/v1/admin/user/${values.id}`;
+      promise = Service.update;
+    }
+    const body = { ...values };
+    promise(url, { body })
+      .then((response: IUser) => {
+        updateData(response);
+      })
+      .catch(() => {
+        updateData(INITIAL_USER_STATE);
```
Comment (rakeshar3796) [inline web/src/page/admin/index.tsx:77]:
```suggestion
        if (modalState === MODAL_STATES.EDIT) {
            updateData(body);
        }
```

File: web/src/page/admin/index.tsx [hunk 5]
```diff
@@ -1,104 +1,309 @@
...
+      });
+    setModalData(INITIAL_USER_STATE);
```
Comment (rakeshar3796) [inline web/src/page/admin/index.tsx:79]: move this to a separate function called resetStates

File: web/src/page/admin/admin.module.css (hunk 1 above)
Comment (muralidharan92) [reply web/src/page/admin/admin.module.css]: updated

File: web/src/page/admin/index.tsx (hunk 2 above)
Comment (muralidharan92) [reply web/src/page/admin/index.tsx]: updated

File: web/src/page/admin/index.tsx (hunk 3 above)
Comment (muralidharan92) [reply web/src/page/admin/index.tsx]: updated

File: web/src/page/admin/index.tsx (hunk 4 above)
Comment (muralidharan92) [reply web/src/page/admin/index.tsx]: updated

File: web/src/page/admin/index.tsx (hunk 5 above)
Comment (muralidharan92) [reply web/src/page/admin/index.tsx]: updated

**Pull request closed without merging**